src/object/cli_obj.c

```diff
@@ -2845,11 +2845,12 @@ obj_shard_list_key_cb(struct shard_auxi_args *shard_auxi,
 		}
 
 		rc = merge_key(&merge_arg->merge_list, key, key_size);
+		/* free key first regardless of rc */
+		if (alloc_key)
```

Comment:
this flag isn't actually necessary.   If key != NULL, it is allocated and D_FREE sets it back to NULL

Reply:
This flag seems more to flag whether key is dynamically allocated here due to oversize needs re-allocate new memory, otherwise key just pointed to existing iov->iov_buf, set this flag to false no no need to free it.